.github/workflows/python_build.yaml

```diff
@@ -0,0 +1,211 @@
+          # and all architectures. If this fails, we will need to update our build strategy to
+          # build separate wheels for each Python version and architecture.
+          WHEEL_NAME=$(ls dist/sift_stack_py*.whl)
+          if [[ ! $WHEEL_NAME =~ "py3-none-any.whl" ]]; then
+            echo "Error: Expected a universal wheel (py3-none-any) but got: $WHEEL_NAME"
+            exit 1
+          fi
+          echo "Verified universal wheel: $WHEEL_NAME"
+
+      - name: Upload distributions
+        uses: actions/upload-artifact@v4
+        with:
+          name: sift-stack-py-dist
+          path: python/dist/*
+          retention-days: 14
+
+  build_and_verify:
+    name: Build offline archive for ${{ matrix.platform.os }} (${{ matrix.platform.arch }}) with Python ${{ matrix.python-version }}
+    needs: [get-matrix-config, build_wheel]
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ${{ fromJson(needs.get-matrix-config.outputs.platforms) }}
+        python-version: ${{fromJson(needs.get-matrix-config.outputs.supported_python_versions)}}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install build tools
+        shell: bash
+        run: |
+          python -m pip install --upgrade pip
+          pip install build pip-tools twine
+
+      - name: Generate requirements
+        working-directory: python
+        shell: bash
+        run: |
+          # Generate requirements file with all extras
+          pip-compile pyproject.toml --all-extras -o requirements-all.txt
+
+      - name: Build dependency wheels
+        working-directory: python
+        shell: bash
+        run: |
+          # Build wheels for all dependencies directly to dist directory
+          # First build wheels for build dependencies
+          pip wheel -w dist \
+            setuptools wheel Cython \
+            --prefer-binary \
+            --no-deps
+
+          # Then build wheels for package dependencies
+          pip wheel -r requirements-all.txt -w dist \
+            --prefer-binary \
+            --no-deps
+
+      - name: Download sift-stack-py wheel
+        uses: actions/download-artifact@v4
+        with:
+          name: sift-stack-py-dist
+          path: python/dist/
+
+      - name: Test installations
+        working-directory: python
+        shell: bash
+        run: |
+          python scripts/build_utils.py \
+            --dist-dir dist \
+            --package-name sift-stack-py \
+            ${{ matrix.platform.os == 'windows' && '--is-windows' || '' }}
+
+      - name: Create distribution archive
+        working-directory: python
+        shell: bash
+        run: |
+          if [ "${{ matrix.platform.os }}" = "windows" ]; then
```

Comment:
lets make tar balls instead of zips, we can publish both in the final merged artifacts